File: footnest_app/lib/models/common/page_response.dart
class PageResponse<T> {

  final List<T> content;

  final int page;

  final int totalPages;

  final int totalElements;

  PageResponse({
    required this.content,
    required this.page,
    required this.totalPages,
    required this.totalElements,
  });

  factory PageResponse.fromJson(
    Map<String, dynamic> json,
    T Function(Map<String, dynamic>) fromJson,
  ) {
    return PageResponse(
      content: (json["content"] as List)
          .map((e) => fromJson(e))
          .toList(),

      page: json["page"],

      totalPages: json["totalPages"],

      totalElements: json["totalElements"],
    );
  }

}

File: footnest-backend/src/main/java/com/footnest/footnest_backend/dto/common/PageResponse.java
package com.footnest.footnest_backend.dto.common;

import lombok.AllArgsConstructor;
import lombok.Getter;

import java.util.List;

@Getter
@AllArgsConstructor
public class PageResponse<T> {

    private List<T> content;


    private int page;


    private int totalPages;


    private long totalElements;

}